### PR TITLE
upgrade TinyLlama to latest checkpoints

### PR DESCRIPTION
**Pull request by aniketmaurya**

### CODE REVIEW

**Comment by Andrei-Aksionov** [comment]
Hello @aniketmaurya 
Each config has two names:
- name: short version (e.g., `tiny-llama-1.1b`)
- long name (HF format) (e.g., `TinyLlama-1.1B-intermediate-step-955k-token-2T`)
https://github.com/Lightning-AI/lit-gpt/blob/0676f23152ee8d7084e18de19339b2422e06641c/lit_gpt/config.py#L1227-L1228

In the PR you are creating multiple configs with different HF names but the same short name.
When `name_to_config` is created, it will keep only one config, the last one.
https://github.com/Lightning-AI/lit-gpt/blob/0676f23152ee8d7084e18de19339b2422e06641c/lit_gpt/config.py#L1280

So, either add a postfix to a short name instead of an empty string, or just keep the latest checkpoint. 
I, personally, prefer the latter.

---

Oh, and could you fix one small issue, that I've missed:
```python
for kind, hf_postfix in (("", "-intermediate-step-955k-token-2T"), ("chat", "-Chat-v0.6")):
```

It should be "-chat" instead.

**Review by aniketmaurya — COMMENTED** [review]
thanks @Andrei-Aksionov, applied your suggestions.

File: lit_gpt/config.py [hunk 1]
```diff
@@ -1243,7 +1243,7 @@ def norm_class(self) -> Type:
     )
 ]
 for c in tiny_llama:
-    for kind, hf_postfix in (("", "-intermediate-step-955k-token-2T"), ("", "-intermediate-step-1431k-3T"), ("chat", "-Chat-v0.6"), ("chat", "-Chat-v1.0")):
+    for kind, hf_postfix in (("", "-intermediate-step-1431k-3T"), ("chat", "-chat-v1.0")):
```
Comment (Andrei-Aksionov) [inline lit_gpt/config.py:1246]: I realized that I've explained it a bit ambiguously 😊 Sorry for that.
Here is what I've meant:

```suggestion
    for kind, hf_postfix in (("", "-intermediate-step-1431k-3T"), ("-chat", "-Chat-v1.0")):
```

File: tutorials/download_tinyllama.md [hunk 2]
```diff
@@ -14,19 +14,19 @@ python scripts/download.py | grep TinyLlama
 which will print
 
 ```text
-TinyLlama/TinyLlama-1.1B-intermediate-step-955k-token-2T
-TinyLlama/TinyLlama-1.1B-Chat-v0.6
+TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+TinyLlama/TinyLlama-1.1B-chat-v1.0
```
Comment (Andrei-Aksionov) [inline tutorials/download_tinyllama.md:18]: Take a look at the changed lines and check that `Chat` is used instead of `chat`.

**Comment by Andrei-Aksionov** [comment]
This one should also be updated.
This test checks that if you retrieve a config by a short name, it will be identical to a config retrieved by a full, HF format, name.

https://github.com/Lightning-AI/lit-gpt/blob/0676f23152ee8d7084e18de19339b2422e06641c/tests/test_config.py#L61

File: tutorials/download_tinyllama.md [hunk 3]
```diff
@@ -14,19 +14,19 @@ python scripts/download.py | grep TinyLlama
...
 ```
 
 In order to use a specific checkpoint, for instance [TinyLlama 1.1B base model](https://huggingface.co/TinyLlama/TinyLlama-1.1B-intermediate-step-955k-token-2T), which requires about 5 GB of disk space, download the weights and convert the checkpoint to the lit-gpt format:
```
Comment (Andrei-Aksionov) [inline tutorials/download_tinyllama.md:21]:
```suggestion
In order to use a specific checkpoint, for instance [TinyLlama 1.1B base model](https://huggingface.co/TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T), which requires about 5 GB of disk space, download the weights and convert the checkpoint to the lit-gpt format:
```

File: tutorials/download_tinyllama.md [hunk 4]
```diff
@@ -58,11 +58,11 @@ python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-interm
...
 # chat version
-python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-Chat-v0.6
+python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-chat-v1.0
 ```
 
 To improve the response from Chat version you can also provide these args (as in the [model card](https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v0.6)):
 
 ```bash
-python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-Chat-v0.6 --top_k=50 --temperature=0.7
+python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-chat-v1.0 --top_k=50 --temperature=0.7
```
[the remark this answers is not in the document]
Comment (aniketmaurya) [reply tutorials/download_tinyllama.md]: updated

File: tutorials/download_tinyllama.md [hunk 5]
```diff
@@ -58,11 +58,11 @@ python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-interm
 or
 
 # chat version
-python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-Chat-v0.6
+python chat/base.py --checkpoint_dir checkpoints/TinyLlama/TinyLlama-1.1B-chat-v1.0
```
[the remark this answers is not in the document]
Comment (aniketmaurya) [reply tutorials/download_tinyllama.md]: updated

**Review by awaelchli — APPROVED** [review]

**Comment by Andrei-Aksionov** [comment]
Good job @aniketmaurya 
The only thing is that you've forgotten about https://github.com/Lightning-AI/lit-gpt/pull/877#issuecomment-1894339236
> It is still in development and at the time of writing this, checkpoints for the model trained up to 2T tokens are available. The target is to train it for ~3 epochs on 3T tokens total.

The checkpoint is trained on 3T tokens.
Though I don't yet understand why it's called an intermediate checkpoint, since 3T tokens was the target 🤷.

Anyway, those sentence in the readme can be changed with the next checkpoint update (if there will be one).

**Pull request merged**